docs/ir/ir-breaking-changes.rst

```diff
@@ -115,6 +115,45 @@ This causes differences in some contracts, for example:
 Previously `f()` would return `0x6465616462656566313564656164000000000000000000000000000000000010` (it has correct length, and correct first 8 elements, but than it contains dirty data which was set via assembly).
 Now it is returning `0x6465616462656566000000000000000000000000000000000000000000000010` (it has correct length, and correct elements, but doesn't contain dirty data).
 
+ * Intermediate values with dirty bits are handled differently in an inline assembly block.
+- The old code generator does not clean up dirty bits of an intermediate value when it is read within an inline assembly block.
+- The new code generator cleans up dirty bits of an intermediate value even when it is read within an inline assembly block.
+
+For example:
+::
+     // SPDX-License-Identifier: GPL-3.0
+     pragma solidity >0.8.0;
+     contract C {
+         function f(uint8 a) public returns(int ret)
+         {
+             a = ~a;
+             assembly {
+                 ret := a
+             }
+         }
+     }
+
+The function `f(uint8)` returns the following for the input 1:
+- Old code gen: `fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe`
+- New code gen: `00000000000000000000000000000000000000000000000000000000000000fe`
+
+* Old code generator evaluates an expression right to left, while the new code gen evaluates an expression left to right. This can lead to semantic differences.
```

Comment:
For the old code generator, the evaluation order is unspecified. For the new code generator, we try to evaluate in source order, but do not guarantee it.

I'm actually not sure if this is also true for function arguments.

Reply:
Updated.